core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -255,7 +257,9 @@ public void watch() {
             }
             notifyCreateEvent(childPath, count, rootPath);
           } else if (kind == ENTRY_MODIFY) {
-            if (fileHasher != null || Files.isDirectory(childPath)) {
+            if (fileHasher == null) {
```

Comment:
I simplified the check here, since my new code depends on the assumption that we only should use the hash code map if hashing is enabled. I'm not sure why we were "hashing" directories to an empty hash even if hashing was disabled.